src/qml/processingparameterwidgets/number.qml

```diff
@@ -0,0 +1,203 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+
+import Theme 1.0
+import org.qfield 1.0
+import org.qgis 1.0
+
+import "."
+
+ProcessingParameterWidgetBase {
+  id: numberItem
+
+  property real step: 1
+  property real min: configuration['minimum']
+  property real max: configuration['maximum']
+  property bool isDouble: configuration['dataType'] === Qgis.ProcessingNumberParameterType.Double
+
+  height: childrenRect.height
+
+  Row {
+    anchors.left: parent.left
+    anchors.right: parent.right
+    anchors.top: parent.top
+
+    TextField {
+      id: textField
+      height: fontMetrics.height + 20
+      topPadding: 10
+      bottomPadding: 10
+      rightPadding: 0
+      leftPadding: enabled ? 5 : 0
+      width: parent.width - decreaseButton.width - increaseButton.width
+
```

Comment:
also in our `coding style` we can declare when to have a `emptyline` and space between our properties. 
I think we should insert a new line when there is a meaningful difference between properties, such as between function declarations.   
removing extra new lines make components smaller, but we have to keep readability.  i think we should talk about this in details later.

Reply:
We can discuss that further yeah.